charts/gateway-helm/templates/certgen.yaml

```diff
@@ -38,12 +38,21 @@ spec:
         resources:
           {{- toYaml . | nindent 10 }}
         {{- end }}
+        securityContext:
+          runAsGroup: 65534
+          runAsUser: 65534
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+            - ALL
       {{- include "eg.image.pullSecrets" . | nindent 6 }}
       restartPolicy: Never
+      {{- with .Values.certgen.job.securityContext }}
```

Comment:
will there be two `securityContext` sections at the same level if a user sets this value ?
is this an `append` or `override` style input field ?

Reply:
No, it won't be two `SecurityContext` on the Pod level, if unset in the Values file, the Pod level `SecurityContext` will be nil, only Container level `SecurityContext` exists.

Example:  Default: https://github.com/envoyproxy/gateway/blob/5e31f2901903992cbde6471fb512dea3ee721ff8/test/helm/gateway-helm/global-images-config.out.yaml
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: envoy-gateway
  namespace: 'envoy-gateway-system'
spec:
  template:
    spec:
      containers:
      - args:
        ....
        securityContext:
          runAsNonRoot: true
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
         ...
      serviceAccountName: envoy-gateway
      terminationGracePeriodSeconds: 10
      volumes:
      - configMap:
          defaultMode: 420
          name: envoy-gateway-config
        name: envoy-gateway-config
      - name: certs
        secret:
          secretName: envoy-gateway
  ```
  With customized `SecurityContex`: https://github.com/envoyproxy/gateway/blob/5e31f2901903992cbde6471fb512dea3ee721ff8/test/helm/gateway-helm/security-context.out.yaml
  
  ```yaml
  apiVersion: apps/v1
kind: Deployment
metadata:
  name: envoy-gateway
  namespace: 'envoy-gateway-system'
spec:
  template:
    spec:
      containers:
      - args:
        ....
        securityContext:
          runAsNonRoot: true
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
         ...
      securityContext:
        fsGroup: 1234
        runAsGroup: 1234
        runAsUser: 1234
      serviceAccountName: envoy-gateway
      terminationGracePeriodSeconds: 10
      volumes:
      - configMap:
          defaultMode: 420
          name: envoy-gateway-config
        name: envoy-gateway-config
      - name: certs
        secret:
          secretName: envoy-gateway
  ```

Reply:
Not sure this is the best way to configure `SecurityContext`, would appreciate your input here @envoyproxy/gateway-maintainers